### PR TITLE
[Merged by Bors] - feat(KrullDimension): height refactor module docstring

### PR DESCRIPTION
**Pull request by nomeata**
This concludes a refactoring of and API extension for `height`. Notable changes, spread over 28 prior PRs, include:

 * Everything put into `namespace Order`
 * Type of `krullDim` changed from `WithBot (WithTop Nat)` to `WithBot ENat`.
 * Type of `height` changed from `WithBot (WithTop Nat)` to `ENat`.
 * Definition of `height` via `LTSeries` rather than krullDim
 * Plenty of lemmas about `height`
 * Concrete calculations for `height` and `krullDim`

This final change updates the module docstring, and adds two `coheight` lemmas corresponding to existing `height` lemmas that slipped through the cracks of juggling multple PRs.

From the Carleson project.

---

Todo:

- [x] Module summary of main results
- [x] docstrings
- [x] `coheight` API
- [x] #15421
- [x] #15385
- [x] #15347
- [x] #15345
- [x] #15342
- [x] #15341
- [x] #15196
- [x] #15387
- [x] #15386
- [x] #15384
- [x] #15344
- [x] #15555
- [x] #15556
- [x] #15557
- [x] #15558
- [x] #15601
- [x] depends on: #16480
- [x] depends on: #16500
- [x] depends on: #16555
- [x] depends on: #19003
- [x] depends on: #19028
- [x] depends on: #19110
- [x] depends on: #19111
- [x] depends on: #19147
- [x] depends on: #19148
- [x] depends on: #19155
- [x] depends on: #19210
- [x] depends on: #19251
- [x] PR some more stuff from other modules

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/KrullDimension.lean
Comment (nomeata) [inline Mathlib/Order/KrullDimension.lean:52]: Ignore this section while reviewing, won't merge this before this is actually empty.

**Review by YaelDillies — APPROVED** [review]
We got to the end of it! Thanks! 🐙 

maintainer merge

(and now we can all go back to doing the hard Carleson analysis)

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
